subprojects/enterprise/src/main/java/org/gradle/internal/enterprise/GradleEnterprisePluginExecutionPhaseListener.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.enterprise;
+
+/**
+ * Used to signal a start of the execution phase to the plugin.
+ *
+ * Expected to be invoked at most once for a build tree, isn't invoked if the configuration phase fails.
+ * The configuration cache doesn't affect this callback, it is invoked regardless of the cache entry being reused.
+ *
+ * Implemented by the Enterprise plugin.
+ *
+ * @see org.gradle.internal.operations.BuildOperationCategory#RUN_MAIN_TASKS
+ */
+public interface GradleEnterprisePluginExecutionPhaseListener {
```

Comment:
Do we need this extra type? Why not inline this on the plugin service?

Reply:
Mostly to follow the existing pattern here and leave room for potential extensions, like adding "executionPhaseEnded" later. I don't have strong feelings, though, and can inline if you prefer that.

Reply:
Let's inline it. It's less busy work to just add more methods to the check in service.

Reply:
👍